packages/serverpod/lib/src/database/database_connection.dart

```diff
@@ -161,115 +63,106 @@ class DatabaseConnection {
         .withInclude(include)
         .build();
 
-    List<TableRow?> list = <TableRow>[];
-    try {
-      var context = transaction != null
-          ? transaction.postgresContext
-          : postgresConnection;
-
-      var result = await context.mappedResultsQuery(
-        query,
-        allowReuse: false,
-        timeoutInSeconds: 60,
-        substitutionValues: {},
-      );
-      for (var rawRow in result) {
-        var rawTableRow = resolvePrefixedQueryRow(
-          table,
-          rawRow,
-          include: include,
-        );
-
-        list.add(poolManager.serializationManager.deserialize<T>(rawTableRow));
-      }
-    } catch (e, trace) {
-      _logQuery(session, query, startTime, exception: e, trace: trace);
-      rethrow;
-    }
-
-    _logQuery(session, query, startTime, numRowsAffected: list.length);
-    return list.cast<T>();
+    return _deserializedMappedQuery<T>(
+      session,
+      query,
+      timeoutInSeconds: 60,
+      transaction: transaction,
+      include: include,
+      table: table,
+    );
   }
 
   /// For most cases use the corresponding method in [Database] instead.
-  Future<T?> findSingleRow<T extends TableRow>({
+  Future<T?> findRow<T extends TableRow>(
```

Comment:
I think they don't really make sense for this type of question, it implies that you need to know that we really are doing a find and then that we take the first value from that list. I think in situations where you want that much control it is better to nudge people to use `find`.

Reply:
I think it would make sense to have `orderBy`, say you want the latest of something or the first. Offset may be less common in this case, but it also doesn't hurt to have it here.

Reply:
I think "Give me the runner with the 3rd fastest time" is a valid question for the findRow interface, and that would require `offset`. It coherent experience for the find interfaces.

But It is also easy to add afterwards so I won't die on this hill :).